### PR TITLE
[CDSK-999] Retry killing a few times, and properly give up if some exception occurs

### PR DESCRIPTION
**Pull request by campos-ddc**

### CODE REVIEW

**Comment by campos-ddc** [comment]
Comitted this because I didn't want to waste some small changes I made.

Could you guys take over this? Create a JIRA task, test in staging, etc.

As mentioned in Slack, we'll also need another task later on to dump the processes being killed, but this one is more important for now.

**Review by Mowinski — APPROVED** [review]
The code looks good :) In future we can use psutil https://pypi.org/project/psutil/#description if we want have code on all platforms

File: slave/buildslave/runprocess.py [hunk 1]
```diff
@@ -821,82 +822,105 @@ def kill(self, msg):
...
+                elif self.process.pid is not None:
+                    logAndSend("using TASKKILL /F PID /T to kill pid %s" % self.process.pid)
+                    for i in range(5):
+                        try:
+                            subprocess.check_call("TASKKILL /F /PID %s /T" % self.process.pid)
+                            logAndSend("taskkill'd pid %s" % self.process.pid)
+                            hit = 1
+                            break
+                        except subprocess.CalledProcessError, e:
+                            if e.returncode == 128:
```
Comment (warcholprzemo) [inline slave/buildslave/runprocess.py:881]: maybe we should set `hit = 1` because we don't know proceed next `if not hit` if "Process no longer exists"

**Review by warcholprzemo — CHANGES_REQUESTED** [review]

File: slave/buildslave/runprocess.py (hunk 1 above)
Comment (campos-ddc) [reply slave/buildslave/runprocess.py:881]: I think you guys would do a better job at maintaining/finishing this, so please make whatever changes you want :)

And yea, it should definitely have a `hit = 1` here

**Review by warcholprzemo — APPROVED** [review]

**Pull request merged**